examples/skyline_selection_results.py

```diff
@@ -0,0 +1,114 @@
+    '''
+        Experiment function to run the experiments with multiple combinations of learners and processors in the input
+    '''
+    exp = GermanCreditDatasetSexExperiment(
+        fixed_random_seed=seed,
+        train_data_sampler=CompleteData(),
+        missing_value_handler=CompleteCaseAnalysis(),
+        numeric_attribute_scaler=NamedStandardScaler(),
+        learners=learners,
+        pre_processors=pre_processors,
+        post_processors=post_processors,
+        optimal_validation_strategy=skyline_strategy)
+    exp.run()
+    return exp.generate_file_path()
+
+def run_exp(seeds, learners, processors, skyline_strategy):
+    '''
+        This is the main driver function that calls the calculate_metrics to give metrices on combinations of various learners, pre and post processing techniques.
+    '''
+    skyline_res_folder = {}
+    for seed in seeds:
+        skyline_res_folder[seed] = calculate_metrics(seed, learners, [x[0] for x in processors], [x[1] for x in processors], skyline_strategy)
+    return skyline_res_folder
+
+
+def output_scatter_plot(f_name, df, x_col, y_col, hue_col='setting', color_p='Set2'):
+    '''
+        Visualization of the skyline options w.r.t. two metrics (X and Y axis) in the skyline inputs.
+    '''
+    sns.set(style='whitegrid', font_scale=1.5)
+    # add jitters for x to account for ties in the values
+    data = df.copy()
+    noise_para = 100
+    data[x_col] += np.random.random(data.shape[0]) / noise_para - 1 / noise_para / 2
+
+    fig, (ax1, ax2) = plt.subplots(1, 2, sharex=True, sharey=True, figsize=(13, 6))
+    sns.scatterplot(x_col, y_col, hue_col, data=data.query("data == 'val'"), ax=ax1, style='optimal', s=100)
+    sns.scatterplot(x_col, y_col, hue_col, data=data.query("data == 'test'"), ax=ax2, style='optimal', s=100)
+    ax1.set_title('validation')
+    ax2.set_title('test')
+    plt.tight_layout()
+
+    # save plot into the disc
+    cur_f_path = f_name[0:f_name.rfind("/") + 1]
+    if not os.path.exists(cur_f_path):
+        directory = os.path.dirname(cur_f_path)
+        pathlib.Path(directory).mkdir(parents=True, exist_ok=True)
+    plt.savefig(f_name + '.png')
+    plt.close()
+
+def get_skyline_candidates(seed_path_map, focus_seed):
+    '''
+        Prepare the skyline candidates data for visualization
+    '''
+    setting_labels = {'reject_option': 'RO', 'diremover-1.0': 'DI1.0',
+                      'no_pre_processing': 'NoPre', 'no_post_processing': 'NoPost',
+                      'DecisionTree': 'DT', 'LogisticRegression': 'LR'}
+    skyline_df = pd.read_csv(seed_path_map[focus_seed] + "skyline_options.csv")
+    # rename the setting of preprocessor (idx 1), learner (idx 5), and postprocessor (idx 6)
+    skyline_df['setting'] = skyline_df['setting'].apply(lambda x: '_'.join([setting_labels[x.split('__')[i].replace('-' + str(focus_seed), '')] for i in range(len(x.split('__'))) if i in [1, 5, 6]]))
```

Comment:
this line is extremely hard to read, whats happening here?

Reply:
Updated to three lines for readability.